src/Modules/Data/Commands.Data.Test/GetPowerBIDatasourceTests.cs

```diff
@@ -91,6 +97,32 @@ public void EndToEndGetPowerBIDatasourceOrganizationScopeAndWorkspaceId()
             }
         }
 
+        [TestMethod]
+        [TestCategory("Interactive")]
+        [TestCategory("SkipWhenLiveUnitTesting")] // Ignore for Live Unit Testing
+        public void EndToEndPipingDatasetIntoGetPowerBIDatasourceIndividualScope()
+        {
+            /*
+             * Requirement to run test:
+             * Need at least one dataset containing a datasource assigned to the user logging into the test.
+             */
+
+            using (var ps = System.Management.Automation.PowerShell.Create())
+            {
+                // Arrange
+                ProfileTestUtilities.ConnectToPowerBI(ps, nameof(PowerBIEnvironmentType.Public));
+                ps.AddCommand(GetPowerBIDatasetTests.GetPowerBIDatasetCmdletInfo).AddCommand(GetPowerBIDatasourceCmdletInfo);
+
+                // Act
+                var results = ps.Invoke();
+
+                // Assert
+                TestUtilities.AssertNoCmdletErrors(ps);
+                Assert.IsNotNull(results);
+                Assert.IsTrue(results.Count > 0);
```

Comment:
Consider using Linq (results.Any()) and possibly asserting inconclusive if no results are found.

Reply:
Added